merklehash/src/lib.rs

```diff
@@ -18,33 +18,30 @@
 //! Generetes complete merkle tree root.
 //!
 //! This module should be used to generate complete merkle tree root hash.
+use std::cmp::PartialEq;
 
-extern crate cita_types;
-extern crate hashable;
-extern crate rlp;
-#[macro_use]
-extern crate rlp_derive;
-
-use cita_types::H256;
-use hashable::{Hashable, HASH_NULL_RLP};
-use rlp::RlpStream;
+pub trait Merge<T> {
+    fn merge(&self, left: &T, right: &T) -> T;
+    fn null(&self) -> T;
```

Comment:
How to remove self?

Reply:
Trait allow defining function inside it. So remove `&self` directly.